api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -299,54 +289,35 @@ public List<org.pmiops.workbench.model.ListRuntimeResponse> deprecatedDeleteRunt
+            deleteRuntime(workspaceNamespace, runtimeNameToDelete)));
   }
 
   @Override
-  public List<AdminRuntimeFields> deleteRuntime(
-      String workspaceNamespace, String runtimeNameToDelete) {
+  public AdminRuntimeFields deleteRuntime(String workspaceNamespace, String runtimeNameToDelete) {
     final String googleProject =
         getWorkspaceByNamespaceOrThrow(workspaceNamespace).getGoogleProject();
-    List<LeonardoListRuntimeResponse> runtimesToDelete =
-        filterByRuntimeName(
-                leonardoNotebooksClient.listRuntimesByProjectAsService(googleProject).stream(),
-                runtimeNameToDelete)
-            .toList();
-    runtimesToDelete.forEach(
-        runtime ->
-            leonardoNotebooksClient.deleteRuntimeAsService(
-                leonardoMapper.toGoogleProject(runtime.getCloudContext()),
-                runtime.getRuntimeName()));
-
-    List<LeonardoListRuntimeResponse> runtimesInProjectAffected =
-        filterByRuntimeName(
-                leonardoNotebooksClient.listRuntimesByProjectAsService(googleProject).stream(),
-                runtimeNameToDelete)
-            .toList();
+    leonardoApiClient.deleteRuntimeAsService(googleProject, runtimeNameToDelete);
+
+    // fetch again to confirm deletion
+    LeonardoGetRuntimeResponse refreshedRuntime =
+        leonardoApiClient.getRuntimeAsService(googleProject, runtimeNameToDelete);
+
     // DELETED is an acceptable status from an implementation standpoint, but we will never
     // receive runtimes with that status from Leo. We don't want to because we reuse runtime
     // names and thus could have >1 deleted runtimes with the same name in the project.
     List<LeonardoRuntimeStatus> acceptableStates =
         List.of(LeonardoRuntimeStatus.DELETING, LeonardoRuntimeStatus.ERROR);
-    runtimesInProjectAffected.stream()
-        .filter(runtime -> !acceptableStates.contains(runtime.getStatus()))
-        .forEach(
-            runtimeInBadState ->
-                log.log(
-                    Level.SEVERE,
-                    String.format(
-                        "Runtime %s/%s is not in a deleting state",
-                        leonardoMapper.toGoogleProject(runtimeInBadState.getCloudContext()),
-                        runtimeInBadState.getRuntimeName())));
-    runtimesToDelete.stream()
-        .map(LeonardoListRuntimeResponse::getRuntimeName)
-        .forEach(
-            runtimeName -> leonardoRuntimeAuditor.fireDeleteRuntime(googleProject, runtimeName));
-    return runtimesInProjectAffected.stream()
-        .map(leonardoMapper::toAdminRuntimeFields)
-        .collect(Collectors.toList());
+    if (!acceptableStates.contains(refreshedRuntime.getStatus())) {
+      log.log(
+          Level.SEVERE,
+          String.format(
+              "Runtime %s/%s is not in a deleting state", googleProject, runtimeNameToDelete));
+    }
+
+    leonardoRuntimeAuditor.fireDeleteRuntime(googleProject, runtimeNameToDelete);
```

Comment:
Would we want to run the auditor if the refreshedRuntime is not in an acceptable state?

Reply:
I think so.  My interpretation of auditor events is that a user requested something, not that it succeeded.